app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt

```diff
@@ -58,7 +61,7 @@ class AutoCompleteApi @Inject constructor(
             BiFunction { bookmarksResults, searchResults ->
                 AutoCompleteResult(
                     query = query,
-                    suggestions = (bookmarksResults + searchResults).distinct()
+                    suggestions = (bookmarksResults + searchResults).distinctBy { it.phrase }
```

Comment:
I'm not 100% sure about this but could be correct. Right now if I have two bookmarks stored with the `editorial.cnn.com` domain and I search by the domain using `editorial` or `cnn` I only get one result back because the domain is used as the phrase and we filter them out here.

I think that both results should be displayed here.

Reply:
So the main problem here is that sealed classes are not value types. And so the distinct does nothing because it is an equality on the instance reference.
I could try to use `data class`  instead

Reply:
Just thinking out loud to summarise what I think we are trying to do:

1. Remove from `searchResults` any items that have a `phrase` which exists in `bookmarkResults`
1. Remove from `bookmarkResults` any items with the same URL.
1. Return 1 + 2

Since everyone is on holiday I'm gonna ask on Asana if that's the intended behaviour or not.

Reply:
Almost. What we want is to dedup the results (1+2).dedup.
I also guess that was the intention of the original `distinc`. However that was not de duplicating anything as it was only comparing instance references